subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -174,13 +223,18 @@ void bl_boot(const struct fw_info *fw_info)
 		"   b   clear\n"
 		"out:\n"
 		"   dsb\n"
+#ifndef CONFIG_SB_DISABLE_SELF_R_X
 		/* Jump to reset vector of an app */
 		"   bx r0\n"
+#endif
 		:
 		: "r" (vector_table[1]), "i" (CONFIG_SRAM_BASE_ADDRESS),
 		  "i" (CONFIG_SRAM_SIZE * 1024), "i" (0)
 		: "r0", "r1", "r2", "r3", "memory"
 	);
+#elif defined(CONFIG_SB_DISABLE_SELF_R_X)
+
+	relocate_bootconf_disable_r_x_and_jump()(vector_table[1]);
```

Comment:
strange syntax. Is that r0 might contain value of reset_vector, if not (vector_table[1])  does the job of populating r0?
I would rather create two alternatives of `relocate_bootconf_disable_r_x_and_jump()` function.

Reply:
Yeah, that might be a problem since at the point of execution of the relocate_bootconf_disable_r_x_and_jump, the vector_table, which is not constant, has already been wiped.
I think the relocate_bootconf_disable_r_x_and_jump should combine both the cleanup and lock.

Reply:
Good catch.

Reply:
it returns pointer to the function in ram that is being called in place with vector_table